android/app/src/main/java/com/vithop/tictactoe/Status.kt

```diff
@@ -1,25 +1,26 @@
 package com.vithop.tictactoe
 
-sealed class Status {
-  object XTurn : Status()
-  object OTurn : Status()
-  data class XWon(val winPositions: Bitboard) : Status() {
-    override fun toString() = super.toString()
-  }
-
-  data class OWon(val winPositions: Bitboard) : Status() {
-    override fun toString() = super.toString()
-  }
+enum class Status {
+  XTurn,
+  OTurn,
+  XWon,
+  OWon,
+  Draw;
 
-  object Draw : Status()
+  private var winPositions = Bitboard()
 
-  override fun toString() = when (this) {
-    XTurn -> "X turn"
-    OTurn -> "O turn"
-    is XWon -> "X won"
-    is OWon -> "O won"
-    Draw -> "Draw"
+  fun getWinPosition(): Bitboard {
+    isWinState()
```

Comment:
Please inline, this is like a contract for the function.
```suggestion
   check(this == XWon || this == OWon) { "winPositions is only valid for status XWon or OWon, status: $this" }
```